app/src/main/java/com/douglasbruce/tasky/features/task/TaskScreen.kt

```diff
@@ -56,21 +66,49 @@ internal fun TaskRoute(
 @Composable
 internal fun TaskScreen(
     onBackClick: () -> Unit,
+    taskUiState: TaskState,
     modifier: Modifier = Modifier,
 ) {
+    val titleDateFormatter = DateTimeFormatter.ofPattern("dd MMMM uuuu")
+    var title = taskUiState.date.format(titleDateFormatter)
+    if (taskUiState.isNew) {
+        title = stringResource(R.string.create_task)
+    } else if (taskUiState.isEditing) {
+        title = stringResource(R.string.edit_task)
+    }
```

Comment:
Probably a matter of taste, but I'm a friend of keeping things a `val` whenever possible. Here, you could do that with a `when` assignment:
```
val title = when {
    taskUiState.isNew -> stringResource(R.string.create_task)
    taskUiState.isEditing -> stringResource(R.string.edit_task)
    else -> taskUiState.date.format(titleDateFormatter)
}
```